File: caesar_cipher.py
'''

                            Online Python Compiler.
                Code, Compile, Run and Debug python program online.
Write your code in this editor and press "Run" button to execute it.

'''

sample_string = "BEWARE THE IDES OF MARCH"
sample_shift = 13

short_shift = sample_shift % 25
encrypted_string = ""

for i in range(0, len(sample_string) - 1) :
    if (sample_string[i] == ' ') :
        encrypted_string += ' '
    elif ( sample_string[i].isupper() ) :
        if( (ord(sample_string[i]) + short_shift > 90) ) :
            encrypted_string += chr( 65 + (short_shift - (90 - ord(sample_string[i]))) - 1 )
        else :
            encrypted_string += chr( ord(sample_string[i]) + short_shift )
    elif ( sample_string[i].islower() ) :
        if( (ord(sample_string[i]) + short_shift > 122) ) :
            encrypted_string += chr( 97 + (short_shift - (122 - ord(sample_string[i]))) - 1)
        else :
            encrypted_string += chr( ord(sample_string[i]) + short_shift )

print("Original message: ", sample_string, "\n")
print("Encrypted message: ", encrypted_string)

File: caesar_cipher.cpp
#include <iostream>
#include <string>
using namespace std;

string caesar_cipher (string original_message, int shift_amount) {
	
	/*
	
	According to the ASCII table:
	- Uppercase letters have the value range of 65 - 90.
	- Lowercase letters have the value range of 97 - 122.
	
	Assume that you only want to change letter characters only.
	
	*/
	
	string encrypted_message = original_message;
	int short_shift = shift_amount;
	
	// shift_amount == 13
	// shift_amount == 87
	// shift_amount == 1000
	
	short_shift = shift_amount % 25;
	
	
	for(int i = 0; i <= encrypted_message.length() - 1; ++i) {
		
		if( (encrypted_message.at(i) == ' ') || (encrypted_message.at(i) == '-') ) { }
		    
		else if(encrypted_message.at(i) >= 'a' || encrypted_message.at(i) <= 'z' || encrypted_message.at(i) >= 'A' || encrypted_message.at(i) <= 'Z') {
		    
		    if(encrypted_message.at(i) + short_shift > 'z' || encrypted_message.at(i) + short_shift > 'Z')
		    {
		        encrypted_message.at(i) = short_shift + int(encrypted_message.at(i)) - 26;
		    }
		    
		    else {
		        encrypted_message.at(i) = encrypted_message.at(i) + short_shift;
		    }
		    
		}
		
	}
	
	return encrypted_message;	
}

int main() {
	
	// string input = "";
	// int shift = 0;
	string sample_string = "BEWARE THE IDES OF MARCH";
	int sample_shift = 13;
	
	// cout << "Enter a message to encrypt: " << endl;
	// cin >> input;
	
	// cout << "Enter the shift amount: " << endl;
	// cin >> shift;
	
	cout << "Original message: " << sample_string << endl;
	cout << "Encrypted message: " << caesar_cipher(sample_string, sample_shift) << endl;
	
	return 0;
}
